x-pack/plugins/apm/server/routes/settings/agent_configuration/route.ts

```diff
@@ -38,7 +45,10 @@ const agentConfigurationRoute = createApmServerRoute({
   ): Promise<{
     configurations: AgentConfiguration[];
   }> => {
+    throwNotFoundIfAgentConfigNotAvailable(resources.settings);
```

Comment:
could it be possible to introduce a centralised mechanism for this?
For example when we are registering the routes

Reply:
you mean as a property on the route?

Reply:
Like knowing from that point () whether the route is enabled or not, and in the case that is not enabled throwing `Boom.notFound()`. `registerRoutes` wouldn't be a good place to do this?